src/printer.rs

```diff
@@ -296,6 +301,18 @@ impl<'a> Printer for InteractivePrinter<'a> {
             }
         }
 
+        // Line highlighting
+        let background = match self.config.highlight_line {
```

Comment:
This whole block could be something like:
``` rust
let background = self.config.highlight_line
    .filter(|line| line == line_number)
    .map(|_| self.background_highlight);
```

Granted, I'm not sure if that is easier to read. Soon, there will be `Option::replace` which could be used instead of `.map(|_| ..)`.